src/common/datomish/query/projection.cljc

```diff
@@ -251,18 +269,31 @@
            elements))))
 
 (defn row-pair-transducer [context]
-  (let [{:keys [elements cc]} context
+  (let [{:keys [find-spec elements cc]} context
         {:keys [source known-types extracted-types]} cc
 
         ;; We know the projection will fail above if these aren't simple variables or aggregates.
         projectors
-        (make-projectors-for-columns elements known-types extracted-types)]
+        (make-projectors-for-columns elements known-types extracted-types)
+
+        single-column-find-spec?
+        (or (instance? FindScalar find-spec)
+            (instance? FindColl find-spec))]
 
     (map
-      (fn [[row err]]
-        (if err
-          [row err]
-          [(map (fn [projector] (projector row)) projectors) nil])))))
+      (if single-column-find-spec?
+        ;; We're only grabbing one result from each row.
+        (let [projector (first projectors)]
```

Comment:
I'm missing something here.  How could you have multiple projectors for when `single-column-find-spec?` is true?  Can you make this stronger, asserting that there's only one?